CMakeLists.txt

```diff
@@ -1246,6 +1246,50 @@ if(WIN32)
   list(APPEND JVM_LIBS -DELAYLOAD:jvm.dll)
 endif(WIN32)
 
```

Comment:
A conflict is expected because we modified and merged to include the openssl-1.1.1f external package into develop branch. Please review up-to-date CMakeFiles.txt files of develop branch.

Reply:
I merged with develop and had no conflicts. I started developing on a branch before openssl was added as an external project, but I found the (openssl as external_project) branch of the pull request and ported it then.

When I finished and I opened the pull request for this, I started from what was then the up-to-date develop (that contained openssl as an external project), I cherry-picked my commits from my old work and I've also removed my addition of the contents of that branch (the openssl). My flow is maybe more complicated than necessary. In this commit I've added it: https://github.com/CUBRID/cubrid/pull/2406/commits/fcb8baf888bc202492fee748310c4bf78d8ae37e and here I've removed it (only when I was cherry-picking the last commit I noticed that openssl as external_project reached develop) https://github.com/CUBRID/cubrid/pull/2406/commits/3aaa04caeb41e4e40ce629f91663c4bdf9cc27a6

Everything should look normal, the portion of adding openssl as an external library is not duplicated and is only what is on develop (I checked with git blame)

Reply:
Sorry to complicate your job. By the way, we have three jobs related with OpenSSL. First one is SSL communication between CAS and drivers, second one is TDE based on OpenSSL and last one is replace gcrypt. And as you can see in CMakeLists.txt, we built openssl in thread mode (build without **no-threads** in **CONFIGURE_COMMAND** for OpenSSL). Is it okay for you?

I think it is well organized and you did good work. But I will continue to review your codes, may be till end of this week

Thank you.

Reply:
Sorry, I wanted to clarify that when I've written "My flow (edited to: my commit set) is maybe more complicated than necessary" I only meant to say the way I structured my commits: that I added the diff from the branch before it was merged and then I removed it. I may have caused a small misunderstanding :smiley: It was no need to apologize for anything, everything is ok.

About the threads flag, it should be ok from this point of view. It even seems that configured with `no-threads`, there could have been problems when accessing openssl low-level internal structures during calls to openssl functions. Reading from the documentation it seems everything is fine now provided we don't call mutating functions on same CTX objects from different threads concurrently.